### ISSUE TITLE
Change DurationNode declaration in Parser

### ISSUE THREAD

**Issue by jsbean** [jsbean/DNM#229]
- Use `DurationNodeStackMode` (`|`, `+`, `-`)  as `openingValue` of `topLevelCommand`
- Add `"Duration"` to allowable types
- Add `"String"` to allowable types for `@`

**Comment by jsbean** [comment]
Create `DurationNodeDeclaration` token.

```Swift
if let durationNodeDeclaration.children.first as? TokenDuration {
    // DurationNodeIntervalEnforced
} else if let durationNodeDeclaration.children.first as? TokenString {
    // if value == "@": DurationNodeIntervalInferred
}
```

**Comment by jsbean** [comment]
Closed with 97c1ae82ae637ed583fb5d4f74f7c573cedd4635
